src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -172,10 +173,97 @@ public async Task GetAsync_SetCookieContainerMultipleCookies_CookiesSent()
             };
         }
 
+        [OuterLoop("Uses delays")]
+        [Fact]
+        public async Task SendAsync_MultipleHttp2ConnectionsEnabled_CreateAdditionalConnections()
+        {
+            // Warm up thread pool because the full .NET Framework calls synchronous Stream.Read() and we need to delay those calls thus threads will get blocked.
+            ThreadPool.GetMinThreads(out _, out int completionPortThreads);
+            ThreadPool.SetMinThreads(401, completionPortThreads);
+            using (WinHttpHandler handler = new WinHttpHandler())
+            {
+                handler.EnableMultipleHttp2Connections = true;
+                handler.ServerCertificateValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+                const int maxActiveStreamsLimit = 100 * 3;
+                const string payloadText = "Multiple HTTP/2 connections test.";
+                TaskCompletionSource<bool> delaySource = new TaskCompletionSource<bool>();
+                using (HttpClient client = new HttpClient(handler))
```

Comment:
nit: reduce indent count with
```suggestion
                using var client = new HttpClient(handler);
```